### PR TITLE
Issue with number of product per page and layered nav refine …

### PR DESCRIPTION
**Pull request by ydmytrunetsspace**

### CODE REVIEW

File: Model/Layer/Filter/Stock.php [hunk 1]
```diff
@@ -75,6 +76,12 @@ public function apply(RequestInterface $request)
             $this->_createItem($this->getLabel($filter), $filter)
         );
 
+        if ($request->getParam($this->getUrlParamName(), null) > 0
+            && $this->isSearchEngineElasticsearch() && $this->isEnabledInStockFilterOnElasticSide()
```
Comment (stkec) [inline Model/Layer/Filter/Stock.php]: if you putting conditions to next line, then to be consistent and increase readability - start all conditions from new line

File: Model/Layer/Filter/Stock.php [hunk 2]
```diff
@@ -75,6 +76,12 @@ public function apply(RequestInterface $request)
             $this->_createItem($this->getLabel($filter), $filter)
         );
 
+        if ($request->getParam($this->getUrlParamName(), null) > 0
```
Comment (stkec) [inline Model/Layer/Filter/Stock.php]: This class has '$this->_requestVar' already initialized with Url Param Name

File: Model/Layer/Filter/Stock.php [hunk 3]
```diff
@@ -167,4 +174,33 @@ public function getProductsCount($value)
...
+        $searchEngine = $this->_scopeConfig->getValue('catalog/search/engine');
+
+        return strpos($searchEngine, 'elasticsearch') !== false;
+    }
+
+    /**
+     * Get url param name
+     * @return string
+     */
+    private function getUrlParamName()
```
Comment (stkec) [inline Model/Layer/Filter/Stock.php]: Redundant function as this class already has $this->_requestVar initialized in the same way in constructor

File: Model/Layer/Filter/Stock.php [hunk 4]
```diff
@@ -167,4 +174,33 @@ public function getProductsCount($value)
...
+    {
+        return $this->_scopeConfig->getValue('s48_stockfilter/settings/url_param');
+    }
+
+    /**
+     * Is enabled in stock filter on Elastic side
```
Comment (stkec) [inline Model/Layer/Filter/Stock.php]: Please avoid useless comments (100% repeats function name)

**Review by stkec — APPROVED** [review]
Please fix code styling (address all comments) before merge

File: Model/Layer/Filter/Stock.php (hunk 1 above)
Comment (ydmytrunetsspace) [reply Model/Layer/Filter/Stock.php]: done

**Pull request merged**